src/navigator/main-stack.tsx

```diff
@@ -21,22 +22,40 @@ import { TokenScreen } from '../screens/token-screen/token-screen';
 import { Wallet } from '../screens/wallet/wallet';
 import { Welcome } from '../screens/welcome/welcome';
 import { useAppLock } from '../shelter/use-app-lock.hook';
-import { useIsAuthorisedSelector } from '../store/wallet/wallet-selectors';
+import { loadActivityGroupsActions } from '../store/activity/activity-actions';
+import { loadSelectedBakerActions } from '../store/baking/baking-actions';
+import { loadTezosBalanceActions, loadTokenBalancesActions } from '../store/wallet/wallet-actions';
+import { useIsAuthorisedSelector, useSelectedAccountSelector } from '../store/wallet/wallet-selectors';
 import { XTZ_TOKEN_METADATA } from '../token/data/tokens-metadata';
 import { emptyTokenMetadata } from '../token/interfaces/token-metadata.interface';
-import { ScreensEnum, ScreensParamList } from './screens.enum';
+import { ScreensEnum, ScreensParamList } from './enums/screens.enum';
+import { useStackNavigatorStyleOptions } from './hooks/use-stack-navigator-style-options.hook';
 import { TabBar } from './tab-bar/tab-bar';
-import { useStackNavigatorStyleOptions } from './use-stack-navigator-style-options.hook';
 
 const MainStack = createStackNavigator<ScreensParamList>();
 
 const isConfirmation = false;
 
 export const MainStackScreen = () => {
+  const dispatch = useDispatch();
   const { isLocked } = useAppLock();
   const isAuthorised = useIsAuthorisedSelector();
+  const selectedAccount = useSelectedAccountSelector();
   const styleScreenOptions = useStackNavigatorStyleOptions();
 
+  useEffect(() => {
+    if (isAuthorised) {
+      const intervalId = setInterval(() => {
```

Comment:
Use recursive `setTimeout` instead of `setInterval`.
Info: https://medium.com/@devinmpierce/recursive-settimeout-8eb953b02b98